spec.bs

```diff
@@ -116,6 +132,49 @@ if (document.wasDiscarded) {
 }
 </pre>
 
+Feature Policies {#feature-policies}
+=====================
+To improve CPU utilization of [=nested browsing contexts=] that are not critical to a page's usability two feature policies are defined:
+ * execution-while-not-rendered, which has a [=default allowlist=] of '*'
+ * execution-while-out-of-viewport, which has a [=default allowlist=] of '*'
+
+The <dfn export>execution-while-not-rendered</dfn> policy controls whether [=task source/freezable=]
+tasks should execute for [=nested browsing contexts=] whose [=browsing context container=] is not [=being rendered=]. ie. `display: none`.
+
+The <dfn export>execution-while-out-of-viewport</dfn> policy controls whether [=task source/freezable=]
+tasks should execute for [=nested browsing contexts=] whose [=browsing context container=] does not intersect the [=viewport=] according
+to [=compute the intersection of a target element and the root=].
+
+Application of these policies is accomplished by placing the document (and its decendants) in a [=/frozen=] state when the following
+conditions have been met:
+ * [=iframe load event steps=] have been run
+ * the policy is disabled for the document
+ * relevant policy condition applies (not rendered or scrolled out of view)
+
+If these conditions aren't met the document will be in the [=/unfrozen=] state.
+
+<pre class="example highlight">
+
+// The iframe will be frozen immediately after it is loaded.
+
+&lt;iframe allow="execution-while-not-rendered 'none'"
+  src="subframe.html" style="display:none"&gt;&lt;/iframe&gt;
+</pre>
+
+To run the <dfn export>update document frozenness steps</dfn> for a {{Document}} <var>document</var>,
+run the following steps:
+1. If <var>document</var>'s [=document/browsing context=] is not a [=nested browsing context=], return.
+1. If <var>document</var>'s [=document/readiness=] is not "complete", return.
+1. Let <var>element</var> be <var>document</var>'s [=nested browsing context=]'s [=browsing context container=].
+1. Let <var>frozenness</var> be unset.
+1. If <var>document</var> is not [=allowed to use=] the [=execution-while-not-rendered=] feature then:
+    1. If <var>element</var>'s is not [=being rendered=],  set <var>frozenness</var>'s to true.
+    1. Otherwise, set <var>frozenness</var> to false.
```

Comment:
I'd set frozenness to false to start and then remove these otherwise steps.

Reply:
Done